stubs/protobuf/google/protobuf/internal/well_known_types.pyi

```diff
@@ -63,29 +67,33 @@ class _FieldMaskTree:
     def AddLeafNodes(self, prefix: tAny, node: tAny) -> None: ...
     def MergeMessage(self, source: tAny, destination: tAny, replace_message: tAny, replace_repeated: tAny) -> None: ...
 
+_StructValue: TypeAlias = struct_pb2.Struct | struct_pb2.ListValue | str | float | bool | None
+_StructValueArg: TypeAlias = _StructValue | Mapping[str, _StructValueArg] | Sequence[_StructValueArg]
+
 class Struct:
-    def __getitem__(self, key: tAny): ...
-    def __contains__(self, item: tAny): ...
-    def __setitem__(self, key: tAny, value: tAny) -> None: ...
-    def __delitem__(self, key: tAny) -> None: ...
+    def __getitem__(self, key: str) -> _StructValue: ...
+    def __contains__(self, item: _StructValueArg) -> bool: ...
+    def __setitem__(self, key: str, value: _StructValueArg) -> None: ...
+    def __delitem__(self, key: str) -> None: ...
     def __len__(self) -> int: ...
-    def __iter__(self): ...
-    def keys(self): ...
-    def values(self): ...
-    def items(self): ...
-    def get_or_create_list(self, key: tAny): ...
-    def get_or_create_struct(self, key: tAny): ...
-    def update(self, dictionary: tAny) -> None: ...
+    def __iter__(self) -> Iterator[str]: ...
+    def keys(self) -> Iterator[str]: ...
+    def values(self) -> Iterator[_StructValue]: ...
+    def items(self) -> Iterator[tuple[str, _StructValue]]: ...
+    def get_or_create_list(self, key: str) -> struct_pb2.ListValue: ...
+    def get_or_create_struct(self, key: str) -> struct_pb2.Struct: ...
+    def update(self, dictionary: Mapping[str, _StructValueArg]) -> None: ...
 
 class ListValue:
     def __len__(self) -> int: ...
-    def append(self, value: tAny) -> None: ...
-    def extend(self, elem_seq: tAny) -> None: ...
-    def __getitem__(self, index: tAny): ...
-    def __setitem__(self, index: tAny, value: tAny) -> None: ...
-    def __delitem__(self, key: tAny) -> None: ...
-    def items(self) -> None: ...
-    def add_struct(self): ...
-    def add_list(self): ...
+    def append(self, value: _StructValue) -> None: ...
+    def extend(self, elem_seq: Iterable[_StructValue]) -> None: ...
+    def __getitem__(self, index: int) -> _StructValue: ...
+    def __setitem__(self, index: int, value: _StructValueArg) -> None: ...
+    def __delitem__(self, key: int) -> None: ...
+    def __iter__(self) -> Iterator[_StructValue]: ...
```

Comment:
I know you added a comment in the allowlist, but I think it wouldn't hurt to add one here as well; it's a little surprising :)

```suggestion
    # Doesn't actually exist at runtime; needed so type checkers understand the class is iterable
    def __iter__(self) -> Iterator[_StructValue]: ...
```